modules/system/assets/js/framework.js

```diff
@@ -30,12 +30,23 @@ if (window.jQuery.request !== undefined) {
         }
 
         /*
-         * Prepare the options and execute the request
+         * Prepare the options
          */
         var $form = options.form ? $(options.form) : $el.closest('form'),
             $triggerEl = !!$form.length ? $form : $el,
             context = { handler: handler, options: options }
 
+        /*
+         * Validate the form client-side
+         */
+        if (!$form[0].checkValidity()) {
```

Comment:
```suggestion
        if ($form[0] && !$form[0].checkValidity()) {
```
`$form` isn't always defined